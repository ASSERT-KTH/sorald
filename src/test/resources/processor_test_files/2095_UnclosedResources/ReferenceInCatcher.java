/*
When an unclosed resource is referenced in a catcher, we must ensure that those references are
removed when the resource is inlined into the resource list.
 */

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class ReferenceInCatcher {

    public static void saveTo(File file) {
        BufferedWriter bw = null;
        try {
            bw = new BufferedWriter(new FileWriter(file)); // Noncompliant
            bw.write("Write some stuff to file");
        } catch (IOException e) {
            e.printStackTrace();
            if (bw != null) {
                bw.close();
            }
        } finally {
            System.out.println("done");
        }
    }
}
