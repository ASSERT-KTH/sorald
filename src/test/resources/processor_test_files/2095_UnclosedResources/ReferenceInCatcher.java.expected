/*
When an unclosed resource is referenced in a catcher, we must ensure that those references are
removed when the resource is inlined into the resource list.
 */

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class ReferenceInCatcher {

    public static void saveTo(File file) throws IOException {
        try (BufferedWriter bw = new BufferedWriter(new FileWriter(file))) {
            bw.write("Write some stuff to file");
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            System.out.println("done");
        }
    }
}
